src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/AttachedCollections/BrowseObjectDescriptionAttributeBase.cs

```diff
@@ -3,18 +3,19 @@
 using System;
 using System.ComponentModel;
 using System.Globalization;
+using System.Resources;
 
 namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.AttachedCollections
 {
     /// <summary>
     /// Specifies a localized description for a property or event.
     /// </summary>
     [AttributeUsage(AttributeTargets.Property)]
-    internal sealed class BrowseObjectDescriptionAttribute : DescriptionAttribute
+    public abstract class BrowseObjectDescriptionAttributeBase : DescriptionAttribute
```

Comment:
I would not make this public.

Reply:
There are three of these base classes which are provided to make implementation easier for users of this system:

- `BrowseObjectBase`
- `BrowseObjectDescriptionAttributeBase`
- `BrowseObjectDisplayNameAttributeBase`

You could argue that `AttachedCollectionItemBase` falls into this category, or even `RelatableItemBase` and `RelationBase<TParent, TChild>`.

Collectively they are a decent amount of code for folks to copy/paste into their projects. Providing them makes it easier to implement the required protocols correctly.

If your concern is that we may wish to change these implementations in future, then one of two things would happen:

- We modify the implementation &mdash; these are abstract base classes which are easier to modify than interfaces.
- We want to use an alternative implementation &mdash; we create new implementations and leave these types in the product.

This PR has 92 new public API members. The attributes add 8. `BrowseObjectBase` adds 5. I would leave them in.